src/server/server/vineyard_server.cc

```diff
@@ -261,6 +262,9 @@ Status VineyardServer::GetData(const std::vector<ObjectID>& ids,
           return status;
         }
       });
+  double endTime = GetCurrentTime();
+  LOG_SUMMARY("data_request_latency", "get", endTime - startTime);
+  LOG_COUNTER("data_requests_total", "get");
```

Comment:
The handlers in socket server runs in an async manner, thus the end time should be moved before the `return status;` in the callback.